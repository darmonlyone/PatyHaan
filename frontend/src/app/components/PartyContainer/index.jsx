import React from "react";

import PartyBox from "app/components/PartyBox";
import useFetchParties from "app/hooks/useFetchParties";

const PartyContainer = ({ history }) => {
	const { parties } = useFetchParties(history);

	return (
		<div className="min-h-screen h-full bg-gray-50 pt-12 lg:pt-16">
			<div className="px-0.5 sm:px-2 lg:px-4 py-2  flex flex-wrap items-center justify-center sm:items-start sm:justify-start">
				{parties.map((party, index) => (
					<PartyBox
						id={party._id}
						key={index}
						imageUrl={party.imageUrl}
						title={party.name}
						amount={party.amount}
						joined={party.userIdList.length}
						status={party.status}
					/>
				))}
			</div>
		</div>
	);
};

export default PartyContainer;
