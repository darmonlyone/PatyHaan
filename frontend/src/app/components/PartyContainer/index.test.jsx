import React from "react";
import { render } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";

import PartyContainer from ".";

jest.mock("axios");

jest.mock("app/hooks/useFetchParties", () => {
	return () => {
		return {
			parties: [
				{
					_id: "60434993f1f8013beab9f687",
					name: "qwqweqweq",
					amount: 4,
					userIdList: ["604344f69d4a6e394d1de227", "60435ffbf1f8013beab9f699"],
					imageUrl: "",
					createAt: "2021-03-06T09:21:23.189Z",
					ownerId: "60428eb421c7a75cb6832c5c",
					status: "none",
				},
			],
		};
	};
});

it("should render PartyContainer with default value", async () => {
	const { asFragment } = render(
		<Router>
			<PartyContainer />
		</Router>
	);

	expect(asFragment()).toMatchSnapshot();
});
