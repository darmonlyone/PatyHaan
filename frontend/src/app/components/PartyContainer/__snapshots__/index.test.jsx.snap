// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render PartyContainer with default value 1`] = `
<DocumentFragment>
  <div
    class="min-h-screen h-full bg-gray-50 pt-12 lg:pt-16"
  >
    <div
      class="px-0.5 sm:px-2 lg:px-4 py-2  flex flex-wrap items-center justify-center sm:items-start sm:justify-start"
    >
      <div
        class="flex flex-col justify-center w-44 lg:w-52 m-1 sm:m-2 lg:m-3 shadow-md border border-gray-100 rounded-md bg-white"
      >
        <img
          alt="No Image"
          class="w-auto h-44 lg:h-52 rounded-t-md"
          data-testid="image"
          src="noimage.png"
        />
        <span
          class="p-2 text-md lg:text-lg max2line h-14 lg:h-16"
        >
          qwqweqweq
        </span>
        <div
          class="border-t border-gray-300 flex flex-row p-2"
        >
          <span
            class="flex w-2/5 text-sm justify-start items-center"
          >
            2/4
          </span>
          <button
            class="w-3/5 text-sm py-1"
          >
            เข้าร่วม
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
